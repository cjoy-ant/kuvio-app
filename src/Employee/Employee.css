@import url("https://fonts.googleapis.com/css2?family=Lora:wght@400;500&display=swap");

.Employee {
  font-family: "Lora", serif;
  font-weight: 400;
  font-size: 0.9em;
  letter-spacing: 0.5px;
}

.Employee__edit-btn {
  background-color: #ffffff;
  color: #1c1c1c;
  border: 0;
  font-size: 24px;
  cursor: pointer;
  margin: 10px 0px 0px 75px;
  transition: 0.4s ease-in-out;
}

.Employee__edit-btn:hover {
  color: #0067d6;
}

.Employee .bold {
  font-weight: bold;
}
