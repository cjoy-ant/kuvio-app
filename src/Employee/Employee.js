import React from "react";
import { Link } from "react-router-dom";
import countryListAllIsoData from "../STORE/countryList";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUserEdit } from "@fortawesome/free-solid-svg-icons";
import { format } from "date-fns";
import { utcToZonedTime } from "date-fns-tz";
import "./Employee.css";

export default class Employee extends React.Component {
  render() {
    const { emp_id, first_name, last_name, country, age, dob } = this.props;

    const capitalizeFirstLetter = (string) => {
      if (string) {
        return string.charAt(0).toUpperCase() + string.slice(1);
      }
    };

    const getCountryName = (countryCode) => {
      if (countryCode) {
        const country = countryListAllIsoData.find(
          (i) => i.code3 === countryCode
        );
        return country.name;
      }
    };

    return (
      <div className="Employee">
        <span className="Employee__name">
          <span className="bold">Name:</span> {capitalizeFirstLetter(last_name)}
          {", "}
          {capitalizeFirstLetter(first_name)}
        </span>
        <br />
        <span className="Employee__country">
          <span className="bold">Country:</span> {getCountryName(country)}
        </span>
        <br />
        <span className="Employee__dob">
          <span className="bold">Date of Birth:</span>{" "}
          {format(utcToZonedTime(dob), "MMMM d, yyyy")}
        </span>
        <br />
        <span className="Employee__age">
          <span className="bold">Age:</span> {age}
        </span>
        <br />
        <Link to={`edit-employee/${emp_id}`}>
          <button className="Employee__edit-btn" type="button">
            <FontAwesomeIcon icon={faUserEdit} alt="edit employee" />
          </button>
        </Link>
      </div>
    );
  }
}
