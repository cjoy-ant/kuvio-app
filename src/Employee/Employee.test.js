import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Employee from "./Employee";

describe(`<Employee />`, () => {
  const props = {
    emp_id: "0bccc578-03c6-11ec-9a03-0242ac130003",
    first_name: "Snapshot",
    last_name: "Test",
    country: "USA",
    age: "30",
    dob: "1990-12-30T00:00:00.000Z",
    date_modified: "2021-08-10T00:00:00.000Z",
  };

  it("renders <Employee /> by default", () => {
    const wrapper = shallow(<Employee />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders <Employee /> given props", () => {
    const wrapper = shallow(<Employee {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
