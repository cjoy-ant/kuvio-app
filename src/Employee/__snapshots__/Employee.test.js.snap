// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Employee /> renders <Employee /> given props 1`] = `
<div
  className="Employee"
>
  <span
    className="Employee__name"
  >
    <span
      className="bold"
    >
      Name:
    </span>
     
    Test
    , 
    Snapshot
  </span>
  <br />
  <span
    className="Employee__country"
  >
    <span
      className="bold"
    >
      Country:
    </span>
     
    United States of America (the)
  </span>
  <br />
  <span
    className="Employee__dob"
  >
    <span
      className="bold"
    >
      Date of Birth:
    </span>
     
    December 29, 1990
  </span>
  <br />
  <span
    className="Employee__age"
  >
    <span
      className="bold"
    >
      Age:
    </span>
     
    30
  </span>
  <br />
  <div
    className="Employee__button-container"
  >
    <Link
      to="edit-employee/0bccc578-03c6-11ec-9a03-0242ac130003"
    >
      <button
        className="Employee__edit-btn"
        type="button"
      >
        <FontAwesomeIcon
          alt="edit employee"
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [],
                "f4ff",
                "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h274.9c-2.4-6.8-3.4-14-2.6-21.3l6.8-60.9 1.2-11.1 7.9-7.9 77.3-77.3c-24.5-27.7-60-45.5-99.9-45.5zm45.3 145.3l-6.8 61c-1.1 10.2 7.5 18.8 17.6 17.6l60.9-6.8 137.9-137.9-71.7-71.7-137.9 137.8zM633 268.9L595.1 231c-9.3-9.3-24.5-9.3-33.8 0l-37.8 37.8-4.1 4.1 71.8 71.7 41.8-41.8c9.3-9.4 9.3-24.5 0-33.9z",
              ],
              "iconName": "user-edit",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </button>
    </Link>
  </div>
</div>
`;
